packages/gestalt/src/TypeaheadInputField.js

```diff
@@ -0,0 +1,207 @@
+// @flow strict
+
+import * as React from 'react';
+import { type Node } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import layout from './Layout.css';
+import styles from './SearchField.css';
+import typeaheadStyle from './TypeaheadInputField.css';
+import Box from './Box.js';
+import Icon from './Icon.js';
+import FormLabel from './FormLabel.js';
+
+type Props = {|
+  id: string,
+  onBlur?: ({
+    event:
+      | SyntheticFocusEvent<HTMLInputElement>
+      | SyntheticKeyboardEvent<HTMLInputElement>,
```

Comment:
@simkessy should we remove `| SyntheticKeyboardEvent<HTMLInputElement>,` here since it is only a `SyntheticFocusEvent<HTMLInputElement>`

Similar as to `onFocus`